pkg/controller/messaginginfra/broker/controller.go

```diff
@@ -0,0 +1,248 @@
+				return err
+			}
+
+			install.ApplyEnvSimple(container, "INFRA_NAME", infra.Name)
+			install.ApplyEnvSimple(container, "CERT_DIR", "/etc/enmasse-certs")
+			install.ApplyEnvSimple(container, "AMQ_NAME", "data")
+			install.ApplyEnvSimple(container, "HOME", "/var/run/artemis")
+
+			// TODO:
+			install.ApplyEnvSimple(container, "ADDRESS_SPACE_TYPE", "shared")
+			install.ApplyEnvSimple(container, "GLOBAL_MAX_SIZE", "-1")
+			install.ApplyEnvSimple(container, "ADDRESS_FULL_POLICY", "FAIL")
+
+			install.ApplyVolumeMountSimple(container, "data", "/var/run/artemis", false)
+			install.ApplyVolumeMountSimple(container, "init", "/opt/apache-artemis/custom", false)
+			install.ApplyVolumeMountSimple(container, "certs", "/etc/enmasse-certs", false)
+			return nil
+		})
+		if err != nil {
+			return err
+		}
+		statefulset.Spec.Template.Spec.InitContainers = initContainers
+
+		containers, err := install.ApplyContainerWithError(statefulset.Spec.Template.Spec.Containers, "broker", func(container *corev1.Container) error {
+			err := install.ApplyContainerImage(container, "broker", infra.Spec.Broker.Image)
+			if err != nil {
+				return err
+			}
+
+			install.ApplyEnvSimple(container, "INFRA_NAME", infra.Name)
+			install.ApplyEnvSimple(container, "CERT_DIR", "/etc/enmasse-certs")
+			install.ApplyEnvSimple(container, "AMQ_NAME", "data")
+			install.ApplyEnvSimple(container, "HOME", "/var/run/artemis")
+			container.Command = []string{"/opt/apache-artemis/custom/bin/launch-broker.sh"}
+
+			// TODO:
+			install.ApplyEnvSimple(container, "ADDRESS_SPACE_TYPE", "shared")
+			install.ApplyEnvSimple(container, "GLOBAL_MAX_SIZE", "-1")
+			install.ApplyEnvSimple(container, "ADDRESS_FULL_POLICY", "FAIL")
+
+			install.ApplyVolumeMountSimple(container, "data", "/var/run/artemis", false)
+			install.ApplyVolumeMountSimple(container, "init", "/opt/apache-artemis/custom", false)
+			install.ApplyVolumeMountSimple(container, "certs", "/etc/enmasse-certs", false)
+
+			return nil
+		})
+		if err != nil {
+			return err
+		}
+		statefulset.Spec.Template.Spec.Containers = containers
+
+		install.ApplyEmptyDirVolume(&statefulset.Spec.Template.Spec, "init")
+		install.ApplySecretVolume(&statefulset.Spec.Template.Spec, "certs", certSecretName)
+
+		statefulset.Spec.VolumeClaimTemplates = []corev1.PersistentVolumeClaim{
+			corev1.PersistentVolumeClaim{
+				ObjectMeta: metav1.ObjectMeta{Name: "data"},
+				Spec: corev1.PersistentVolumeClaimSpec{
+					AccessModes: []corev1.PersistentVolumeAccessMode{
+						corev1.ReadWriteOnce,
+					},
+
+					Resources: corev1.ResourceRequirements{
+						Requests: map[corev1.ResourceName]resource.Quantity{"storage": *resource.NewScaledQuantity(2, resource.Giga)},
+					},
+				},
+			},
+		}
+		return nil
+	})
+
+	if err != nil {
+		return err
+	}
+
+	_, err = b.certController.ReconcileCert(ctx, logger, infra, statefulset)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func int32ptr(v int32) *int32 {
```

Comment:
I would love to have this in a more global package!

Reply:
IMO this is such a small utility and something that will never change, so keeping it close to the code where it is used increases readability, and there is no question about maintainability in this case.

Reply:
No, however you already have the same function twice, and I know at least two more locations with similar code in the IoT parts … wouldn't it make sense to have a single function?